### PR TITLE
fix(release): allow releases from release branch

### PR DESCRIPTION
**Pull request by nellyk**

### CODE REVIEW

**Review by Francois-Esquire — DISMISSED** [review]

File: lerna.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
...
   "command": {
     "version": {
-      "conventionalCommits": true,
-      "allowBranch": "main"
+      "conventionalCommits": true
     }
-  }
-}
+  },
+  "version": "independent"
```
Comment (Francois-Esquire) [inline lerna.json:11]:
```suggestion
  }
```

**Review by mtomcal — APPROVED** [review]

**Comment by JamesSingleton** [comment]
@nellyk, before renaming to `main`, we were allowing `master`: https://github.com/americanexpress/holocron/blob/5b3a5787dea29ce77621801883a4666db2b9abb4/lerna.json. What is the need for now allowing releases from any branch?

**Comment by nellyk** [comment]
> @nellyk, before renaming to `main`, we were allowing `master`: https://github.com/americanexpress/holocron/blob/5b3a5787dea29ce77621801883a4666db2b9abb4/lerna.json. What is the need for now allowing releases from any branch?

Since we run the release process on branches on branches that we create locally, removing this would allow us to run `lerna:version` on any branch. If this is set to main we would only be able to run the command on the main branch

**Comment by JamesSingleton** [comment]
> > @nellyk, before renaming to `main`, we were allowing `master`: https://github.com/americanexpress/holocron/blob/5b3a5787dea29ce77621801883a4666db2b9abb4/lerna.json. What is the need for now allowing releases from any branch?
> 
> Since we run the release process on branches on branches that we create locally, removing this would allow us to run `lerna:version` on any branch. If this is set to main we would only be able to run the command on the main branch. Having the master branch here also would have caused the same problem thus we were unable to do any releases

Versioning sure, but releasing should only be done on a specific branch. That is why we still open the PR to `main` to kick off the release.

**Comment by nellyk** [comment]
> > > @nellyk, before renaming to `main`, we were allowing `master`: https://github.com/americanexpress/holocron/blob/5b3a5787dea29ce77621801883a4666db2b9abb4/lerna.json. What is the need for now allowing releases from any branch?
> > 
> > 
> > Since we run the release process on branches on branches that we create locally, removing this would allow us to run `lerna:version` on any branch. If this is set to main we would only be able to run the command on the main branch. Having the master branch here also would have caused the same problem thus we were unable to do any releases
> 
> Versioning sure, but releasing should only be done on a specific branch. That is why we still open the PR to `main` to kick off the release.

When running `lerna:version` locally it checks first that the branch is on the list of allowed branches and in this case if you've created a branch with the name `chore/release` the below process would fail. 
 >Run yarn lerna:version locally from your release branch. This would push your release changes(changelog and tags) to the branch on github. https://github.com/americanexpress/holocron/blob/main/CONTRIBUTING.md

**Comment by JamesSingleton** [comment]
@nellyk so maybe we need to do something like this https://github.com/vercel/next.js/blob/canary/lerna.json where the publishing is only allowed on certain branches

**Comment by nellyk** [comment]
> > > @nellyk, before renaming to `main`, we were allowing `master`: https://github.com/americanexpress/holocron/blob/5b3a5787dea29ce77621801883a4666db2b9abb4/lerna.json. What is the need for now allowing releases from any branch?
> > 
> > 
> > Since we run the release process on branches on branches that we create locally, removing this would allow us to run `lerna:version` on any branch. If this is set to main we would only be able to run the command on the main branch. Having the master branch here also would have caused the same problem thus we were unable to do any releases
> 
> Versioning sure, but releasing should only be done on a specific branch. That is why we still open the PR to `main` to kick off the release.

This would work if we allow users to push directly to protected branches(master and canary). If we specify this it would create an overhead and we do usually delete the branches afterwards and it doesn't affect the code quality e.g https://github.com/americanexpress/one-app-cli/pull/136 i beleive it would be overkill to require release branches names to follow a certain format.

**Comment by JAdshead** [comment]
> > > > @nellyk, before renaming to `main`, we were allowing `master`: https://github.com/americanexpress/holocron/blob/5b3a5787dea29ce77621801883a4666db2b9abb4/lerna.json. What is the need for now allowing releases from any branch?
> > > 
> > > 
> > > Since we run the release process on branches on branches that we create locally, removing this would allow us to run `lerna:version` on any branch. If this is set to main we would only be able to run the command on the main branch. Having the master branch here also would have caused the same problem thus we were unable to do any releases
> > 
> > 
> > Versioning sure, but releasing should only be done on a specific branch. That is why we still open the PR to `main` to kick off the release.
> 
> This would work if we allow users to push directly to protected branches(master and canary) and trigger a release. If we specify this it would create an overhead and we do usually delete the branches afterwards and it doesn't affect the code quality/process e.g [americanexpress/one-app-cli#136](https://github.com/americanexpress/one-app-cli/pull/136) i believe it would be overkill to require release branches names to follow a certain format.

are we able to do a branch pattern ? `chore/release-*`

**Review by Francois-Esquire — DISMISSED** [review]

**Comment by nellyk** [comment]
> > > > > @nellyk, before renaming to `main`, we were allowing `master`: https://github.com/americanexpress/holocron/blob/5b3a5787dea29ce77621801883a4666db2b9abb4/lerna.json. What is the need for now allowing releases from any branch?
> > > > 
> > > > 
> > > > Since we run the release process on branches on branches that we create locally, removing this would allow us to run `lerna:version` on any branch. If this is set to main we would only be able to run the command on the main branch. Having the master branch here also would have caused the same problem thus we were unable to do any releases
> > > 
> > > 
> > > Versioning sure, but releasing should only be done on a specific branch. That is why we still open the PR to `main` to kick off the release.
> > 
> > 
> > This would work if we allow users to push directly to protected branches(master and canary) and trigger a release. If we specify this it would create an overhead and we do usually delete the branches afterwards and it doesn't affect the code quality/process e.g [americanexpress/one-app-cli#136](https://github.com/americanexpress/one-app-cli/pull/136) i believe it would be overkill to require release branches names to follow a certain format.
> 
> are we able to do a branch pattern ? `chore/release-*`

yes this is possible

> > > > > @nellyk, before renaming to `main`, we were allowing `master`: https://github.com/americanexpress/holocron/blob/5b3a5787dea29ce77621801883a4666db2b9abb4/lerna.json. What is the need for now allowing releases from any branch?
> > > > 
> > > > 
> > > > Since we run the release process on branches on branches that we create locally, removing this would allow us to run `lerna:version` on any branch. If this is set to main we would only be able to run the command on the main branch. Having the master branch here also would have caused the same problem thus we were unable to do any releases
> > > 
> > > 
> > > Versioning sure, but releasing should only be done on a specific branch. That is why we still open the PR to `main` to kick off the release.
> > 
> > 
> > This would work if we allow users to push directly to protected branches(master and canary) and trigger a release. If we specify this it would create an overhead and we do usually delete the branches afterwards and it doesn't affect the code quality/process e.g [americanexpress/one-app-cli#136](https://github.com/americanexpress/one-app-cli/pull/136) i believe it would be overkill to require release branches names to follow a certain format.
> 
> are we able to do a branch pattern ? `chore/release-*`
Updated it to match this. thanks for the suggestions @JamesSingleton  and @JAdshead

**Review by JAdshead — APPROVED** [review]

**Review by mtomcal — APPROVED** [review]

**Review by Francois-Esquire — APPROVED** [review]

**Pull request merged**